3rdparty/ymfm/src/ymfm.h

```diff
@@ -325,6 +325,86 @@ struct ymfm_output
 };
 
 
+// ======================> ymfm_wavfile
+
+// this class is a debugging helper that accumulates data and writes it to wav files
+template<int _Channels>
+class ymfm_wavfile
+{
+public:
+	// construction
+	ymfm_wavfile(uint32_t samplerate = 44100) :
+		m_samplerate(samplerate)
+	{
+	}
+
+	// configuration
+	ymfm_wavfile &set_index(uint32_t index) { m_index = index; return *this; }
+	ymfm_wavfile &set_samplerate(uint32_t samplerate) { m_samplerate = samplerate; return *this; }
+
+	// destruction
+	~ymfm_wavfile()
+	{
+		if (!m_buffer.empty())
+		{
+			// create file
+			char name[20];
+			sprintf(name, "wavlog-%02d.wav", m_index);
+			FILE *out = fopen(name, "wb");
+
+			// make the wav file header
+			uint8_t header[44];
+			memcpy(&header[0], "RIFF", 4);
+			*(uint32_t *)&header[4] = m_buffer.size() * 2 + 44 - 8;
+			memcpy(&header[8], "WAVE", 4);
+			memcpy(&header[12], "fmt ", 4);
+			*(uint32_t *)&header[16] = 16;
+			*(uint16_t *)&header[20] = 1;
+			*(uint16_t *)&header[22] = _Channels;
+			*(uint32_t *)&header[24] = m_samplerate;
+			*(uint32_t *)&header[28] = m_samplerate * 2 * _Channels;
+			*(uint16_t *)&header[32] = 2 * _Channels;
+			*(uint16_t *)&header[34] = 16;
+			memcpy(&header[36], "data", 4);
+			*(uint32_t *)&header[40] = m_buffer.size() * 2 + 44 - 44;
+
+			// write header then data
+			fwrite(&header[0], 1, sizeof(header), out);
+			fwrite(&m_buffer[0], 2, m_buffer.size(), out);
+			fclose(out);
+		}
+	}
+
+	// add data to the file
+	template<int _Outputs>
+	void add(ymfm_output<_Outputs> output)
```

Comment:
Another case of using a reserved name beginning with an underscore followed by an uppercase letter.